### ISSUE TITLE
SourceBranchName is not the full name. Troublesome for scheduling Jenkins (multibranch pipeline) jobs

### ISSUE THREAD

**Issue by daantimmer** [Microsoft/vsts-tasks#8193]
## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting
- Not required

## Environment
- Server - VSTS or TFS on-premises?
Microsoft Visual Studio Team Foundation Server
Version 16.122.27102.1

- Agent - Hosted or Private: 
N/A

## Issue Description
When creating Job definitions where a Jenkins job is supposed to be scheduled and/or downloading from Jenkins is supposed to be done the provided variables for the branch naming is inconvenient.
The current variable Build.SourceBranchName is removing all the prefixes before the last forward slash. So a branch named 'feature/my-shiny-new-feature' is reduced to 'my-shiny-new-feature'.
This is cumbersome when combining vsts and Jenkins Multibranch Pipeline feature as there is now a need to provide this prefix manually. That also means a build definition per prefix name.
Example for the shiny-new-feature branch mentioned above the `Multibranch pipeline Branch` name is set to: `feature/$(Build.SourceBranchName)`. Now if I would also have branches starting with `spike/**` and `experimental/**` I would have to make two more build definitions.

As proposed here: https://github.com/Microsoft/vsts-agent/issues/838 there seems to be a need for an extra variable exposing the whole branch name including the prefix.

### Error logs
N/A

**Comment by CraigHead** [comment]
@ericsciple I have and am currently using this as a work-around. Just feels like I shouldn't have to do that. Also, to answer your question to @daantimmer, if I'm understanding the architecture I feel like this is something wrong with the service, which is populating this variable and sending it down to agent tasks.

**Comment by daantimmer** [comment]
@ericsciple to me it sounds like an issue with the service not exposing the "correct"/wanted variable.

**Comment by vtbassmatt** [comment]
You're right - it would be awesome if Build.SourceBranchName reflected everything after `refs/heads/`. Unfortunately, that's not what it does. I'm not willing to take the compat break here: the current behavior isn't totally insane, we know of customers using the current behavior, and there's a trivial workaround.

When we get a chance to revisit how we inject variables into the task process, we will definitely do the more obvious thing here :)

**Comment by daantimmer** [comment]
@vtbassmatt but would it be a big issue of adding a third variable? For this exact purpose. No need to break compatibility.

**Comment by vtbassmatt** [comment]
@daantimmer we're already unhappy with the large quantity of variables we've introduced over the years (and can never take away for compat reasons). We're being much more judicious about introducing new ones today.

I also just re-read the issue more carefully and noticed you're on TFS. Even if we added a variable, it likely wouldn't land in a TFS update. You'd have to wait for the next major version (now called Azure DevOps Server).
